### PR TITLE
feat(Heroku): add presence

### LINKED ISSUE #5275: Heroku | heroku.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4655

<div type='discussions-op-text'>

<sup>Originally posted by **Rxann** May 24, 2021</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 

**What's the name of the service?** 
https://heroku.com | Heroku

**What should the Presence display?** 
The presence should display That you are visiting the Heroku site, and if you are making an app it should display that.

**If possible, please provide a logo for the service (512x512 minimum)** 

logo
</div>

### PR DESCRIPTION
**Pull request by theusaf**
## Description 
Adds a Presence for Heroku.
Resolves #5275

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/H/Heroku/presence.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
+const presence = new Presence({
+		clientId: "1014999001801691246",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1e3);
```
Comment (RisingSunLight42) [inline websites/H/Heroku/presence.ts:4]: Why making something simple as complicated.
```suggestion
	browsingTimestamp = Math.floor(Date.now() / 1000);
```

File: websites/H/Heroku/presence.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+					}
+					case "/account/applications": {
+						presenceData.state = "Application settings";
+						break;
+					}
+					case "/account/billing": {
+						presenceData.state = "Billing settings";
+						break;
+					}
+					// No default
```
Comment (RisingSunLight42) [inline websites/H/Heroku/presence.ts:57]: Useless, we can see it by ourself
```suggestion

```

File: websites/H/Heroku/presence.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+							presenceData.state = "Activity";
+							if (subpath2 === "builds")
+								presenceData.state = "Viewing build log";
+							break;
+						}
+						default: {
+							presenceData.state = `${subpath[0].toUpperCase()}${subpath.slice(
+								1
+							)}`;
+						}
```
Comment (RisingSunLight42) [inline websites/H/Heroku/presence.ts:79]: No sure for this, but an if is maybe better there, the switch add more indent that can be annoying to read.
@EncryptedDev what do you think ?

File: websites/H/Heroku/presence.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+			}
+			break;
+		}
+		case "data.heroku.com": {
+			presenceData.details = "Browsing Data";
+			if (pathname === "/") presenceData.state = "Looking at datastores";
+			else if (pathname === "/dataclips")
+				presenceData.state = "Looking at dataclips";
+			else if (pathname.startsWith("/datastores/")) {
+				if (await presence.getSetting<boolean>("showNames")) {
```
Comment (RisingSunLight42) [inline websites/H/Heroku/presence.ts:96]: You should store the setting in a variable at the top of the updateData func. You use it multiple time.

**Review by RisingSunLight42 — CHANGES_REQUESTED** [review]

File: websites/H/Heroku/presence.ts (hunk 3 above)
Comment (theusaf) [reply websites/H/Heroku/presence.ts:79]: That makes sense. This was mostly left over from when a switch made more sense, but I refactored it and removed unneeded cases, but forgot to convert it to an if statement

**Review by RisingSunLight42 — APPROVED** [review]

File: websites/H/Heroku/presence.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+				);
+				presenceData.state = `Apps: ${appStatus.textContent.replace(/\s/g, "")}
+Data: ${dataStatus.textContent.replace(/\s/g, "")}
+Tools: ${toolsStatus.textContent.replace(/\s/g, "")}`;
+			}
+			break;
+		}
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
```
Comment (DarkVillager) [inline websites/H/Heroku/presence.ts:160]:
```suggestion
	if (presenceData.details) presence.setActivity(presenceData);
	else presence.setActivity();
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/H/Heroku/presence.ts (hunk 5 above)
Comment (EncryptedDev) [reply websites/H/Heroku/presence.ts:160]: Not necessarily needed

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**